datafusion/physical-plan/src/aggregates/mod.rs

```diff
@@ -487,10 +483,10 @@ impl AggregateExec {
         ));
         let original_schema = Arc::new(original_schema);
         // Reset ordering requirement to `None` if aggregator is not order-sensitive
-        order_by_expr = aggr_expr
+        let mut order_by_expr = aggr_expr
```

Comment:
This is the key observation, right? That the ordering can be derived from the `AggregateExprs` themselves

Reply:
Exactly